packages/api-fetch/src/test/index.js

```diff
@@ -183,4 +183,39 @@ describe( 'apiFetch', () => {
 			} );
 		} );
 	} );
+
+	it( 'should not use the default fetch handler when using a custom fetch handler', () => {
+		const customFetchHandler = jest.fn();
+
+		apiFetch.setFetchHandler( customFetchHandler );
+
+		apiFetch( { path: '/random' } );
+
+		expect( window.fetch ).not.toHaveBeenCalled();
+
+		expect( customFetchHandler ).toHaveBeenCalledWith( {
+			path: '/random?_locale=user',
+		} );
+	} );
+
+	it( 'should run the last-registered user-defined middleware first', () => {
+		// This could potentially impact other tests in that a lingering
+		// middleware is left. For the purposes of this test, it is sufficient
+		// to ensure that the last-registered middleware receives the original
+		// options object. It also assumes that some built-in middleware would
+		// either mutate or clone the original options if the extra middleware
+		// had been pushed to the stack.
+		expect.assertions( 2 );
+
+		const expectedOptions = {};
+
+		apiFetch.use( ( actualOptions, next ) => {
+			expect( actualOptions ).toBe( expectedOptions );
+			expect( actualOptions ).toEqual( expectedOptions );
```

Comment:
From https://jestjs.io/docs/en/expect#tobevalue:

>Use `.toBe` to compare primitive values or to check referential identity of object instances. It calls `Object.is` to compare values, which is even better for testing than `===` strict equality operator.

Nit: I think the 2nd check  (`. toEqual`) is obsolete in this test.

Reply:
You're right. `toEqual` is useful for to check the "deep" equality of two different objects. Removed in e2dc3d3

Reply:
> From https://jestjs.io/docs/en/expect#tobevalue:
> 
> > Use `.toBe` to compare primitive values or to check referential identity of object instances. It calls `Object.is` to compare values, which is even better for testing than `===` strict equality operator.
> 
> Nit: I think the 2nd check (`. toEqual`) is obsolete in this test.

The idea was to check that _neither_ the reference changed _nor_ the value mutated. I think `toBe` could return `true` even if the value was directly mutated?

```
var a = {}
var b = a
b.foo = 1
Object.is( a, b )
// true
```

Reply:
`b.foo = 1` also implies `a.foo = 1`, so if the reference is the same, the value will be the same:

```js
const a = {};
const b = a;
b.foo = 1;

expect( a ).toBe( b ); // true
expect( a ).toEqual( b ); // true
expect( a.foo ).toEqual( 1 ); // true
```

Reply:
I'd thought there could be some case where one is not sufficient to capture both conditions, but on reflection I think `toBe` should be enough. 👍